### PR TITLE
feat: Adding Python SDK for git step with Python3-9 runtime.

### PR DESCRIPTION
**Pull request by domderen**
This SDK is implementing the image contract(https://github.com/argoproj-labs/argo-dataflow/blob/main/docs/IMAGE_CONTRACT.md), by:
- Responding to /ready with a 204 status code once it is able to start serving requests,
- Accepting POST request to /messages, and responding with 201, 204 or 500 accordingly,
- Handles SIGTERM gracefully, giving server time to finish processing the request before terminating.

### CODE REVIEW

**Comment by alexec** [comment]
Wow! Great stuff. When I wrote the Galang SDK, I updated runtimes/Galang-16 to use the SDK. Do you think we should do the same for this?

**Comment by domderen** [comment]
Yeah, I'm doing it right now, give me a moment.

**Comment by domderen** [comment]
Ok, so I added an example showing how to use this in a git step with a python runtime. I also added this:

```Dockerfile
RUN pip3 install git+https://github.com/argoproj-labs/argo-dataflow#subdirectory=sdks/python
```

To steps building the `python3-9` image. But I'm not sure where are python files inside this docker image, so I could modify them to use the new SDK. Can you point me in the right direction?

File: examples/106-git-pipeline.py [hunk 1]
```diff
@@ -6,13 +6,18 @@
...
-your code when the step starts.
+your code when the step starts. This example presents how one can use both go runtime & python runtime git steps.
 
 [Learn about Git steps](../docs/GIT.md)""")
      .step(
         (kafka('input-topic')
          .git('main', 'https://github.com/argoproj-labs/argo-dataflow', 'main', 'examples/git', 'golang:1.16',
               env=[{'name': "GOCACHE", 'value': "/tmp/.gocache"}], command=["go", "run", "."])
-         .kafka('output-topic')
+         .kafka('go-output-topic')
```
Comment (alexec) [inline examples/106-git-pipeline.py:16]: Can we have a second separate Python example?

File: examples/106-git-pipeline.py [hunk 2]
```diff
@@ -6,13 +6,18 @@
...
+         ))
+     .step(
+        (kafka('input-topic')
+         .git('main', 'https://github.com/argoproj-labs/argo-dataflow', 'main', 'examples/git-python', 'python:3.9', command=["./start.sh"])
+         .kafka('python-output-topic')
```
Comment (alexec) [inline examples/106-git-pipeline.py:21]: Examples are tested to make sure they can run. You’ll need to stick with input-topic and output-topic

File: examples/106-git-pipeline.py (hunk 1 above)
Comment (domderen) [reply examples/106-git-pipeline.py:16]: As in `106-git-python-pipeline.py` that is a copy of the other one with a different runtime?

File: Dockerfile [hunk 3]
```diff
@@ -89,6 +89,7 @@ ADD runtimes/python3-9 /workspace
 RUN chown -R 9653 /.cache /.local /workspace
 WORKDIR /workspace
 USER 9653:9653
+RUN pip3 install git+https://github.com/argoproj-labs/argo-dataflow#subdirectory=sdks/python
```
Comment (alexec) [inline Dockerfile:92]: Should this go into requirements.txt?

Comment (domderen) [reply Dockerfile:92]: It should but I'm not sure where this requirements.txt lives. I couldn't find files that go into this image...

**Comment by alexec** [comment]
I could merge this as the code needs to be on main to be built and pass test?

File: sdks/python/argo_dataflow_sdk/main.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    return self.keepRunning
+
+  def terminate(self, signal, frame):
+    print("Start Terminating: %s" % datetime.now(), signal)
+    self.keepRunning = False
+
+  def start(self, handler):
+    signal.signal(signal.SIGTERM, self.terminate)
+    MyServer.handler = handler
+    self.webServer = HTTPServer((hostName, serverPort), MyServer)
```
Comment (domderen) [inline sdks/python/argo_dataflow_sdk/main.py:48]: I'm wondering if we should use `ThreadingHTTPServer` rather than `HTTPServer`?

**Comment by domderen** [comment]
Yeah, I think you can go for it.

**Comment by alexec** [comment]
🚀

**Pull request merged**